wp-includes/rest-api/endpoints/class-wp-rest-customize-controls-controller.php

```diff
@@ -0,0 +1,371 @@
+
+		$control = $wp_customize->get_control( $request['control'] );
+		if ( ! $control ) {
+			return new WP_Error( 'rest_control_invalid_id', __( 'Invalid control ID.' ), array(
+				'status' => 403,
+			) );
+		}
+
+		return $control->check_capabilities();
+	}
+
+	/**
+	 * Retrieves a single control.
+	 *
+	 * @since 4.?.?
+	 * @access public
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_item( $request ) {
+		$wp_customize = $this->ensure_customize_manager();
+
+		$control = $wp_customize->get_control( $request['control'] );
+
+		return rest_ensure_response( $this->prepare_item_for_response( $control, $request ) );
+	}
+
+	/**
+	 * Checks if a given request has access to read controls.
+	 *
+	 * @since 4.?.?
+	 * @access public
+	 *
+	 * @param  WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True if the request has read access, WP_Error object otherwise.
+	 */
+	public function get_items_permissions_check( $request ) {
+		return current_user_can( 'edit_theme_options' );
+	}
+
+	/**
+	 * Retrieves list of controls.
+	 *
+	 * @since 4.?.?
+	 * @access public
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_items( $request ) {
+		$wp_customize = $this->ensure_customize_manager();
+
+		$all_controls = $wp_customize->controls();
+		$controls = array();
+
+		foreach ( $all_controls as $control_id => $control ) {
+			if ( ! $control->check_capabilities() ) {
+				continue;
+			}
+			$data = $this->prepare_item_for_response( $control, $request );
+			$controls[] = $this->prepare_response_for_collection( $data );
+		}
+
+		return rest_ensure_response( $controls );
+	}
+
+	/**
+	 * Prepares a single control's output for response.
+	 *
+	 * @since 4.?.?
+	 * @access public
+	 *
+	 * @param WP_Customize_Control $control WP_Customize_Control object.
+	 * @param WP_REST_Request      $request Request object.
+	 * @return WP_REST_Response Response object.
+	 */
+	public function prepare_item_for_response( $control, $request ) {
+
+		$control_array = (array) $control;
+		$data = array();
+		$primary_setting = '';
+
+		$links = array(
+			'self' => array(
+				'href' => rest_url( trailingslashit( $this->namespace . '/' . $this->rest_base ) . $control_array['id'] ),
+			),
+		);
+
+		$hide_from_response = array(
+			'capability',
+			'instance_number',
+			'manager',
+			'json',
+			'setting', // Hide this from response since that's returned within 'settings'.
+			'active_callback',
+		);
+
+		// Remove unused params of upload control.
+		if ( $control instanceof WP_Customize_Upload_Control ) {
+			$hide_from_response[] = 'removed';
+			$hide_from_response[] = 'context';
+			$hide_from_response[] = 'extensions';
+		}
```

Comment:
^ @westonruter Added here the logic for removing the unused params of `WP_Customize_Upload_Control`.